prow/plugins/yuks/yuks.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yuks
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/plugins"
+
+	"net/http"
+)
+
+var (
+	match  = regexp.MustCompile(`(?m)^@k8s-bot tell me a joke$`)
+	simple = regexp.MustCompile(`^[\w?.,]`)
+)
+
+const (
+	jokeUrl    = realJoke("http://tambal.azurewebsites.net/joke/random")
+	pluginName = "yuks"
+)
+
+func init() {
+	plugins.RegisterIssueCommentHandler(pluginName, handleIssueComment)
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	AddLabel(owner, repo string, number int, label string) error
+	RemoveLabel(owner, repo string, number int, label string) error
+}
+
+type joker interface {
+	readJoke() (string, error)
+}
+
+type realJoke string
+
+func (url realJoke) readJoke() (string, error) {
```

Comment:
Should really add a simple test using `httptest` that this does what you think it does.